### PR TITLE
Background layer switcher

### PR DESCRIPTION
**Pull request by fschmenger**
This branch adds a background layer switcher control to the map according to #203.

Wegue_bglayerswitcher

A description on how to configure background layers and preview images is available from the updated documentation.

### CODE REVIEW

File: app-starter/static/app-conf-projected.json [hunk 1]
```diff
@@ -86,9 +86,8 @@
       "projection": "EPSG:28992",
       "tileGridRef": "dutch_rd",
       "attributions": "<a href='https://www.pdok.nl' target='_blank'>PDOK</a> by Dutch Kadaster",
-      "isBaseLayer": false,
-      "visible": false,
-      "displayInLayerList": true
+      "isBaseLayer": true,
```
Comment (chrismayer) [inline app-starter/static/app-conf-projected.json:89]: This seems not be an appropriate base layer, this has more an overlay character, so I vote set `isBaselayer: false` here

File: src/components/bglayerswitcher/LayerPreviewImage.vue [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      return this.layer.get('previewImage') || LayerPreview.getUrl(
+        this.layer,
+        this.mapView.getCenter(),
+        this.mapView.getResolution(),
+        this.mapView.getProjection()
+      );
+    }
+  }
+};
+</script>
```
Comment (chrismayer) [inline src/components/bglayerswitcher/LayerPreviewImage.vue:47]: Please add a new line at the end of files

**Review by chrismayer — APPROVED** [review]
This is really cool, thanks for this @fschmenger! I like the component a lot. Some questions for clarification:

- Is there a way to hide / disable the bgLayerSwitcher somehow? Or ist it always visible as long as you have one base layer?
- Custom positioning et. al. has to be done by CSS?

Could you please add some basic unit tests? This would be very much appreciated.

**Review by chrismayer — APPROVED** [review]
This is really cool, thanks for this @fschmenger! I like the component a lot. Some questions for clarification:

- Is there a way to hide / disable the bgLayerSwitcher somehow? Or ist it always visible as long as you have one base layer?
- Custom positioning et. al. has to be done by CSS?

Could you please add some basic unit tests? This would be very much appreciated.

**Comment by fschmenger** [comment]
Hi Christian, I just returned from a 2 weeks holiday. Thanks for the review, I will look into the requested changes and provide some basic unit tests. Regarding your questions.

* The switcher becomes automatically visible, as soon as 1 base layer is configured.
* Positioning uses CSS (for now). This is obviously not one of the generic modules, as the behavior and placement of the button is custom.

**Comment by fschmenger** [comment]
@chrismayer All requested changes should have been addressed now.

**Comment by chrismayer** [comment]
Thanks for your ongoing effort on this @fschmenger! Regarding your comments:

> The switcher currently is always visible and I haven't introduced any config option to disable it. 

OK for me at least for the moment. Let's have a look how often it is requested not to have the BackgroundSwitchter tool. As a workaround the button could be hidden by CSS in a Wegue application. 

> Positioning uses CSS (for now). This is obviously not one of the generic modules, as the behavior and placement of the button is custom.

Thanks for clarification. This is totally OK and it should be tackled in a bigger view, how to place modules, et. al.

I'll merge now.

**Pull request merged**